### PR TITLE
Make HTTP Listen Address configurable

### PR DESCRIPTION
**Pull request by cburgdorf**
### What was wrong?

As explained in #1795, Trinity does currently not allow accessing its JSON RPC HTTP API from an external address. This should be configureable.

### How was it fixed?

1. Exposed the listen address as `--http-listen-address` but kept the default at `127.0.0.1`

2. Fixed some casing inconsistencies in the help texts (Server > server)

3. Renamed `--rpcport` to `--http-port` to be more aligned and precise (RPC doesn't necessarily refer to the HTTP API as it is an umbrella term that includes RPC access via HTTP, IPC and potentially WS in the future) 

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: trinity/components/builtin/json_rpc/component.py [hunk 1]
```diff
@@ -103,15 +103,21 @@ def configure_parser(cls, arg_parser: ArgumentParser, subparser: _SubParsersActi
...
         arg_parser.add_argument(
             "--enable-http",
             action="store_true",
-            help="Enables the HTTP Server",
+            help="Enables the HTTP server",
         )
         arg_parser.add_argument(
-            "--rpcport",
+            "--http-listen-address",
+            type=str,
```
Comment (cburgdorf) [inline trinity/components/builtin/json_rpc/component.py:115]: I initially thought we would want this to be of type `ipaddress` but geth allows this to be bound to a string and `HTTPServer` exposes this as a string, too so I guess binding to some DNS name might be a valid thing (I have no idea, I'm not a networking expert)

**Comment by cburgdorf** [comment]
@hwwhww pinged you for a review since you've last worked on this and I'm also renaming a flag that you might be using.

**Comment by hwwhww** [comment]
@cburgdorf  Hi!
Other files that are using `--rpcport` flags need to be updated too:
1. https://github.com/ethereum/trinity/blob/48188a06d9b9df79eae6dd17b559f77c08e5e600/docs/api/api.cli.rst
2. https://github.com/ethereum/trinity/blob/84381f4fe19b18dddf2d2b210bd0780038728acd/eth2/beacon/scripts/run_beacon_nodes.py#L133

**Comment by cburgdorf** [comment]
@hwwhww Ooops, sloppy me! Thanks for pointing me to these places. Should be fixed!

**Pull request merged**